packages/databricks-vscode/src/bundle/bundleAutocompleteProvider.ts

```diff
@@ -18,21 +24,22 @@ export async function generateBundleSchema(cli: CliWrapper) {
 
         // We use the API exposed from teh activate() function of the redhat.vscode-yaml
         // extension to registor a custom schema provider
+        let bundleFileList = await bundleFileSet.allFiles();
+        context.subscriptions.push(
+            bundleWatcher.onDidChangeRootFile(async () => {
+                bundleFileList = await bundleFileSet.allFiles();
+            })
+        );
         redHatYamlSchemaApi.registerContributor(
             "dabs",
             (resource: string) => {
-                const validFileNames: string[] = [
-                    "databricks.yml",
-                    "databricks.yaml",
-                    "bundle.yml",
-                    "bundle.yaml",
-                ];
-                for (const name of validFileNames) {
-                    if (path.basename(resource) === name) {
-                        return rootConfigSchemaUri;
-                    }
+                if (
+                    bundleFileList.find(
+                        (i) => i.fsPath === Uri.parse(resource).fsPath
```

Comment:
This can call `Uri.parse` outside of the lambda to save some cycles.

Reply:
Updated here as well https://github.com/databricks/databricks-vscode/pull/922